modin/numpy/arr.py

```diff
@@ -0,0 +1,1459 @@
+        result._query_compiler = result._query_compiler.astype(
+            {col_name: out.dtype for col_name in result._query_compiler.columns}
+        )
+        if isinstance(where, array):
+            out._query_compiler = where.where(result, out)._query_compiler
+        elif where:
+            out._query_compiler = result._query_compiler
+        return out
+    if isinstance(where, array) and out is None:
+        from array_creation import zeros_like
+
+        out = zeros_like(result).astype(dtype if dtype is not None else result.dtype)
+        out._query_compiler = where.where(result, out)._query_compiler
+        return out
+    elif not where:
+        from .array_creation import zeros_like
+
+        return zeros_like(result)
+    return result
+
+
+def find_common_dtype(dtypes):
+    if len(dtypes) == 1:
+        return dtypes[0]
+    elif len(dtypes) == 2:
+        return numpy.promote_types(*dtypes)
+    midpoint = len(dtypes) // 2
+    return numpy.promote_types(
+        find_common_dtype(dtypes[:midpoint]), find_common_dtype(dtypes[midpoint:])
+    )
+
+
+class array(object):
+    """
+    Modin distributed representation of ``numpy.array``.
+
+    Internally, the data can be divided into partitions along both columns and rows
+    in order to parallelize computations and utilize the user's hardware as much as possible.
+
+    Notes
+    -----
+    The ``array`` class is a lightweight shim that relies on the pandas Query Compiler in order to
+    provide functionality.
+    """
+
+    def __init__(
+        self,
+        object=None,
+        dtype=None,
+        *,
+        copy=True,
+        order="K",
+        subok=False,
+        ndmin=0,
+        like=numpy._NoValue,
+        _query_compiler=None,
+        _ndim=None,
+    ):
+        if _query_compiler is not None:
+            self._query_compiler = _query_compiler
+            self._ndim = _ndim
+            new_dtype = find_common_dtype(
+                numpy.unique(self._query_compiler.dtypes.values)
+            )
+        elif is_list_like(object) and not is_list_like(object[0]):
+            series = pd.Series(object)
+            self._query_compiler = series._query_compiler
+            self._ndim = 1
+            new_dtype = self._query_compiler.dtypes.values[0]
+        else:
+            target_kwargs = {
+                "dtype": None,
+                "copy": True,
+                "order": "K",
+                "subok": False,
+                "ndmin": 0,
+                "like": numpy._NoValue,
+            }
+            for key, value in target_kwargs.copy().items():
+                if value == locals()[key]:
+                    target_kwargs.pop(key)
+                else:
+                    target_kwargs[key] = locals()[key]
+            arr = numpy.array(object, **target_kwargs)
+            assert arr.ndim in (
+                1,
+                2,
+            ), "Modin.NumPy currently only supports 1D and 2D objects."
```

Comment:
```suggestion
            ), "`modin.numpy` currently only supports 1D and 2D objects."
```

Reply:
Do we want to lowercase everything, or keep NumPy's capitalization? @devin-petersohn @noloerino

Reply:
I agree with lowercasing since this implicitly is referring to `modin.numpy` as a python module, rather than NumPy as a tool.

Reply:
Fixed!